### PR TITLE
Allows custom modules to be added to Snowball

### PR DESCRIPTION
**Pull request by brianlee389**

### CODE REVIEW

File: inc/snowball-editor.php [hunk 1]
```diff
@@ -122,8 +150,8 @@
     'templates'   => $templates,
     'names'       => $names,
     'adminUrl'    => admin_url(),
+    'pluginsUrl'  => plugins_url("snowball"),
     'includesUrl' => includes_url(),
-    'pluginsUrl'  => plugins_url('', dirname(__FILE__)),
```
Comment (thomaspark) [inline inc/snowball-editor.php:126]: Looks like you didn't fetch the latest `develop` before rebasing. I changed this in an earlier commit to get rid of this hardcoded path we missed.

File: snowball.php [hunk 2]
```diff
@@ -12,6 +12,14 @@
...
 $path = plugin_dir_path(__FILE__);
 
+function create_custom_modules_directory() {
+	$cm_dir_path = dirname(dirname(__FILE__)) . "/custommodules";
+	if (!file_exists($cm_dir_path)) {
+		mkdir($cm_dir_path, 0755);
+	}
+}
+register_activation_hook( __FILE__, 'create_custom_modules_directory');
+
```
Comment (thomaspark) [inline snowball.php]: We don't need this function anymore, do we?

File: inc/snowball-editor.php [hunk 3]
```diff
@@ -76,17 +87,34 @@
...
+      $type = basename($module);
+      $blocks[$type] = file_get_contents($module . '/admin.html');
+      $templates[$type] = file_get_contents($module . '/template.html');
+
+      $json = file_get_contents($module . '/admin.json');
+      $meta = json_decode($json);
+
+      $names[$type] = $meta->name;
+      $iconClasses[$type] = $meta->iconClasses;
+      $order[] = strtolower($type);
```
Comment (thomaspark) [inline inc/snowball-editor.php]: Is `strtolower()` needed here? It gets done again in line 129.

**Comment by brianlee389** [comment]
I'm sorry I thought I made the change with the custom modules. Also strtolower is not needed I'll remove that

**Comment by thomaspark** [comment]
Looks good, but appears you changed line 153:

```
'pluginsUrl'  => plugins_url('', dirname(__FILE__)),
```

back to the hard-coded:

```
'pluginsUrl'  => plugins_url("snowball"),
```

If you don't have your dev machine anymore, no problem, I can make the changes myself.

**Pull request merged**